test/widgets/actions_test.dart

```diff
@@ -30,19 +41,247 @@ void main() {
+        await store.addMessage(message);
+        await tester.pump();
+      }
+
+      addTearDown(testBinding.reset);
+
+      final account1 = eg.account(id: 1, user: eg.user());
+      final account2 = eg.account(id: 2, user: eg.user());
+      await testBinding.globalStore.add(account1, eg.initialSnapshot());
+      await testBinding.globalStore.add(account2, eg.initialSnapshot());
+
+      final testNavObserver = TestNavigatorObserver();
+      await tester.pumpWidget(ZulipApp(navigatorObservers: [testNavObserver]));
+      await tester.pump();
+      final navigator = await ZulipApp.navigator;
+      navigator.popUntil((_) => false); // clear starting routes
+      await tester.pumpAndSettle();
+
+      final pushedRoutes = <Route<dynamic>>[];
+      testNavObserver.onPushed = (route, prevRoute) => pushedRoutes.add(route);
+      final account1Route = InboxPage.buildRoute(accountId: account1.id);
+      final account2Route = InboxPage.buildRoute(accountId: account2.id);
+      unawaited(navigator.push(account1Route));
+      unawaited(navigator.push(account2Route));
+      await tester.pumpAndSettle();
+      check(pushedRoutes).deepEquals([account1Route, account2Route]);
+
+      await makeUnreadTopicInInbox(account1.id, 'topic in account1');
+      final findAccount1PageContent = find.text('topic in account1', skipOffstage: false);
+
+      await makeUnreadTopicInInbox(account2.id, 'topic in account2');
+      final findAccount2PageContent = find.text('topic in account2', skipOffstage: false);
+
+      final findLoadingPage = find.byType(LoadingPlaceholderPage, skipOffstage: false);
+
+      check(findAccount1PageContent).findsOne();
+      check(findLoadingPage).findsNothing();
+
+      final removedRoutes = <Route<dynamic>>[];
+      testNavObserver.onRemoved = (route, prevRoute) => removedRoutes.add(route);
+
+      final context = tester.element(find.byType(MaterialApp));
+      final future = logOutAccount(context, account1.id);
+      await tester.pump(TestGlobalStore.removeAccountDuration);
+      await future;
+      check(removedRoutes).single.identicalTo(account1Route);
+      check(findAccount1PageContent).findsNothing();
+      check(findLoadingPage).findsOne();
+
+      await tester.pump();
+      check(findAccount1PageContent).findsNothing();
+      check(findLoadingPage).findsNothing();
+      check(findAccount2PageContent).findsOne();
+    });
+  });
+
+  group('unregisterToken', () {
+    testWidgets('smoke, happy path', (tester) async {
+      await prepare(tester, ackedPushToken: '123');
+
+      final newConnection = separateConnection()
+        ..prepare(json: {'msg': '', 'result': 'success'});
+      final future = unregisterToken(testBinding.globalStore, eg.selfAccount.id);
+      await tester.pump(Duration.zero);
+      await future;
+      assertSingleUnregisterRequest(newConnection, expectedToken: '123');
+      check(newConnection.isOpen).isFalse();
+    });
```

Comment:
```suggestion
    }, variant: const TargetPlatformVariant({TargetPlatform.android, TargetPlatform.iOS}));
```

That way we exercise both the Android and iOS cases in unregisterToken.

Reply:
(I think adding this to just this test is enough.)